### ISSUE TITLE
Syntax error in type: string[]=  (jsdoc/valid-types)

### ISSUE THREAD

**Issue by yvele** [gajus/eslint-plugin-jsdoc#494]
Writing an optional array parameter with Google Closure Compiler syntax

```js
/**
 * @param {string[]=} foo Description.
 */
```

Is returning an error

```
error  Syntax error in type: string[]=  jsdoc/valid-types
```

But `{string[]=}` looks valid 🤔 the `=` character should not be part of the "type"

I'm using

```js
settings : {
  jsdoc : {
    preferredTypes : {
      "Array.<>" : "[]",
      ".<>" : "<>"
    }
  }
},
rules : {
   "jsdoc/check-syntax": "off"
}

```

**Comment by brettz9** [comment]
Does Closure accept it? I see at https://jsdoctypeparser.github.io/ that using `(string[])=` works though I don't know if that should be necessary.

If this should work, please file an issue at https://github.com/jsdoctypeparser/jsdoctypeparser , with indication of whether it works in Closure (and/or if you can find it, whether the docs make clear it is allowed).

**Comment by yvele** [comment]
Ok I've created an issue in the parser repo https://github.com/jsdoctypeparser/jsdoctypeparser/issues/110

**Comment by brettz9** [comment]
Just saw your comment about not using Closure itself now, but if you can at least link to the docs for the optional syntax so we can clarify whether it should work without parentheses, that should be helpful. Thanks...

**Comment by brettz9** [comment]
I think we can close and track there now instead. Thanks again!

**Comment by yvele** [comment]
@brettz9 I cannot find any mention of an optional notation of `value[]` using Closure 🤔 

But the real problem for me is the `fix` of `preferredTypes: { "Array.<>" : "[]" }` that converts me

```js
/**
 * @param {Array.<string>=} foo
 */
```

to

```js
/**
 * @param {string[]=} foo
 */
```

And the `fix` turns out to convert an error to another error

The code should be fixed to

```js
/**
 * @param {(string[])=} foo
 */
```

Do you want me to open a new issue?

**Comment by yvele** [comment]
As you stated in [the parser issue](https://github.com/jsdoctypeparser/jsdoctypeparser/issues/110#issuecomment-590624574) the parenthesis are unnecessary as the `{string[]=}` notation is perfectly explicit without any ambiguity.

As I understand when the parser issue https://github.com/jsdoctypeparser/jsdoctypeparser/issues/110 will be fixed there will be no need to wrap `{string[]=}` around parenthesis during fix.

**Comment by brettz9** [comment]
Ok, closing then for now and can track there.
